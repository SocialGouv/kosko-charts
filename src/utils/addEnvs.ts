import type { Deployment } from "kubernetes-models/apps/v1/Deployment";
import { EnvVar } from "kubernetes-models/v1/EnvVar";

import { addEnv } from "./addEnv";

type AnyObject = Record<string, string>;

/** Parameters for the [[addEnvs]] function */
interface AddEnvsParams {
  /** [[Deployment]] to add env vars for */
  deployment: Deployment;
  /** an object with all env vars to add */
  data: AnyObject;
  /** The container index for which to add the env var, default=0 */
  containerIndex?: number;
}

/**
 *
 * This function will add a bunch of [[EnvVar]] to the first container of a given [[Deployment]]
 * EnvVars are given as an object
 *
 * ```typescript
 * import { addEnvs } from "@socialgouv/kosko-charts/utils"
 *
 * addEnvs({
 *   deployment,
 *   data: {
 *     NODE_ENV: "production",
 *     TEST: "42"
 *   }
 * })
 * ```
 * @category utils
 * @return {void}
 */
export const addEnvs = ({
  deployment,
  data,
  containerIndex = 0,
}: AddEnvsParams) => {
  Object.keys(data).forEach((key) => {
    addEnv({
      containerIndex,
      data: new EnvVar({ name: key, value: data[key] }),
      deployment,
    });
  });
};
