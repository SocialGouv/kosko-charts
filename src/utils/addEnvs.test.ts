import { Deployment } from "kubernetes-models/apps/v1/Deployment";

import { addEnvs } from "./addEnvs";

test("should add HELLO and ANSWER variables in deployment env", () => {
  const deployment = new Deployment({
    metadata: { name: "test1" },
    spec: {
      selector: {},
      template: { spec: { containers: [{ name: "container1" }] } },
    },
  });
  addEnvs({ data: { ANSWER: "42", HELLO: "world" }, deployment });
  expect(deployment).toMatchSnapshot();
});
