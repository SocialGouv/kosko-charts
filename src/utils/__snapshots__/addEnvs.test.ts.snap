// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add HELLO and ANSWER variables in deployment env 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "metadata": Object {
    "name": "test1",
  },
  "spec": Object {
    "selector": Object {},
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "env": Array [
              Object {
                "name": "ANSWER",
                "value": "42",
              },
              Object {
                "name": "HELLO",
                "value": "world",
              },
            ],
            "name": "container1",
          },
        ],
      },
    },
  },
}
`;
